### ISSUE TITLE
Can not use mocks package for testing

### ISSUE THREAD

**Issue by astrekalova** [Shopify/sarama#552]
I just updated the import of sarama to the new version. I use the "gopkg.in/Shopify/sarama.v1" import and it works in my script for sending out Kafka messages, however I have a problem using "gopkg.in/Shopify/sarama.v1/mocks" in my test. I get an exception: 
go build code.google.com/p/snappy-go/snappy: no buildable Go source files in 

Is there a solution for that?

**Comment by eapache** [comment]
This may be a duplicate of #497 - make sure you update your copy of `github.com/Shopify/sarama` as well

**Comment by astrekalova** [comment]
Sorry, I am confused what package I need to use for testing with mocks - "gopkg.in/Shopify/sarama.v1/mocks" or "github.com/Shopify/sarama/mocks". I had a problem with "github.com/Shopify/sarama" with the google snappy dependency, that's why I moved to "gopkg.in/Shopify/sarama.v1/mocks".

**Comment by eapache** [comment]
Due to limitations of go's import system, you are basically stuck using `github.com/Shopify/sarama/mocks` (this is the root of #497).

The problem you are seeing in this ticket is probably the same problem you were seeing with using `github.com/Shopify/sarama` directly - the google snappy dependency changed the layout of their package and so you have to upgrade it in parallel with sarama to maintain compatibility.

`go get -u github.com/Shopify/sarama` may be enough to upgrade everything. If not it might be simpler to blow away your copy of sarama and snappy and try cloning them again from scratch.

**Comment by astrekalova** [comment]
After I upgraded with ```go get -u github.com/Shopify/sarama``` I get an error: 
./main_test.go:35: cannot use dataCollectorMock (type *mocks.AsyncProducer) as type "gopkg.in/Shopify/sarama.v1".AsyncProducer in field value:
	*mocks.AsyncProducer does not implement "gopkg.in/Shopify/sarama.v1".AsyncProducer (wrong type for Errors method)
		have Errors() <-chan *"github.com/Shopify/sarama".ProducerError
		want Errors() <-chan *"gopkg.in/Shopify/sarama.v1".ProducerError

Do I have to update anything else? Do I have to use ```github.com/Shopify/sarama``` in main.go as well?

**Comment by eapache** [comment]
Ya, *this* is the #497 issue now: if you want to use mocks, you have to use `github.com/Shopify/sarama` everywhere because go can't resolve the imports otherwise
